tests/language-feature/enums/zero-initialize/regular.slang

```diff
@@ -0,0 +1,64 @@
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -xslang -zero-initialize
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-vk -compute -entry computeMain -emit-spirv-directly -allow-glsl -xslang -zero-initialize
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-cpu -compute -entry computeMain -allow-glsl -xslang -zero-initialize
+//TEST(smoke,compute):COMPARE_COMPUTE_EX(filecheck-buffer=BUF):-dx12 -use-dxil -compute -entry computeMain -allow-glsl -xslang -zero-initialize
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<int> outputBuffer;
+
+enum MyFlags
+{
+    Zero = 0,
+    One,
+}
```

Comment:
I am wondering if we should also test cases of enum that inherited from integer types.
Something like,
```
enum MyEnum : uint16_t // <=== all integer types like int8_t and int64_t
{ Whatever };
MyEnum var;
Test if var is zero.
```

Reply:
I can add it, never a problem to add more.